SFS_ir/get_ir.m

```diff
@@ -124,28 +140,53 @@
     % For SimpleFreeFieldHRIR only the relative position between listener
     % position and source position is of relevance.
     xs = xs-X+X_sofa;
-    % Get measured loudspeaker positions and go to spherical coordinates
-    x0 = sofa_get_secondary_sources(header,'spherical');
-    [xs(1),xs(2),xs(3)] = cart2sph(xs(1),xs(2),xs(3));
+    % Get measured loudspeaker positions
+    x0 = sofa_get_secondary_sources(header,'cartesian');
+    x0 = x0(:,1:3); % need position only
     % Combine head orientation and desired direction of source (see note above)
+    [xs(1),xs(2),xs(3)] = cart2sph(xs(1),xs(2),xs(3));
     xs(1) = correct_azimuth(xs(1)-head_orientation(1));
     xs(2) = correct_elevation(xs(2)-head_orientation(2));
-    % Find nearest neighbours and interpolate if desired and needed
-    [neighbours,idx] = findnearestneighbour(x0(:,1:2)',xs(1:2),3);
+    [xs(1),xs(2),xs(3)] = sph2cart(xs(1),xs(2),xs(3));
+    % Get nearest neighbour point or points for interpolation
+    if useinterpolation
+        switch interpolationpointselection
+        case 'nearestneighbour'
+            [idx,weights] = findnearestneighbour(x0,xs,2);
+        case 'delaunay'
+            [idx,weights] = findconvexcone(x0,xs);
+        case 'voronoi'
+            to_be_implemented;
+        otherwise
+            error(['%s: ''%s'' is an unknown method to select interpolation ', ...
+                'points.'],upper(mfilename),interpolationpointselection);
+        end
+    else
+        [idx,weights] = findnearestneighbour(x0,xs,1);
+    end
+    % Get impulse responses according to point selection
     ir = sofa_get_data_fir(sofa,idx);
+    % Correct distance
+    [x0(:,1),x0(:,2),x0(:,3)] = cart2sph(x0(:,1),x0(:,2),x0(:,3));
+    [xs(1),xs(2),xs(3)] = cart2sph(xs(1),xs(2),xs(3));
     ir = ir_correct_distance(ir,x0(idx,3),xs(3),conf);
-    [ir,x0] = interpolate_ir(ir,neighbours,xs(1:2)',conf);
-    [x0(1),x0(2),x0(3)] = sph2cart(x0(1),x0(2),xs(3));
+    [x0(:,1),x0(:,2),x0(:,3)] = sph2cart(x0(:,1),x0(:,2),x0(:,3));
+    % Select or interpolate to desired impulse response
+    [ir,weights_selected,x0_selected] = interpolate_ir(ir,weights,x0(idx,:),conf);
+    % Calculate position of returned impulse response
+    x0_new = weights_selected'*x0_selected;
```

Comment:
Hm, I know that part is not very pretty. The problem is, that with the new calculation of weights for points used for interpolation, a linear combination of the weighted points does not yield the desired new point but it lies on the direct connection between two points or on the plane spanned by three points. That is, the radius is not what was intended. Only the direction is relevant for the interpolation weights. Adjusting of the amplitude due to a different source to listener distance is done by `ir_correct_distance()`.
And as I thought the calculation of `x0_new` with fixing of the radius should not be inside of `interpolate_ir()`, I calculated it this way.